torch/csrc/jit/interpreter.cpp

```diff
@@ -638,10 +669,20 @@ struct CodeImpl {
 // InterpreterState state that and used to compute a Code
 struct InterpreterStateImpl {
   InterpreterStateImpl(const Code & code)
-  : function(code.pImpl),
-    int_data(function->int_data.data()),
-    bool_data(function->bool_data),
-    registers(function->register_size) {
+    : function(code.pImpl),
+      int_data(function->int_data.data()),
+      bool_data(function->bool_data),
+      registers(function->register_size) {
+    loadConstantsIntoRegisters();
+  }
+
+  void loadConstantsIntoRegisters() {
+    for (auto constant: function->constants) {
+      Value * val = constant.first;
+      IValue ivalue = constant.second;
+      int reg = function->getOrAllocateRegister(val, true);
```

Comment:
This is in the fast path, so it is not a good idea to do hash table lookups. Simply make the first C registers all the constants, then do a  a  std::copy of a flat vector of constants into the registers.